api/src/main/java/io/minio/MinioClient.java

```diff
@@ -968,10 +1066,26 @@ private void updateRegionCache(String bucketName)
       }
 
       // Add the new location.
-      BucketRegionCache.INSTANCE.add(bucketName, region);
+      BucketRegionCache.INSTANCE.set(bucketName, region);
     }
   }
 
+  /**
+   * Computes region of a given bucket name. If set, this.region is considered. Otherwise,
+   * resort to the server location API.
+   */
+  private String bucketToRegion(String bucketName) throws InvalidBucketNameException, NoSuchAlgorithmException,
```

Comment:
I think `getRegion()` is more appropriate than `bucketToRegion()` which looks like a conversion method.